framework/include/functions/PiecewiseBilinear.h

```diff
@@ -63,6 +63,11 @@ class PiecewiseBilinear : public Function
   virtual Real value(Real t, const Point & pt) const override;
   virtual ADReal value(const ADReal & t, const ADPoint & pt) const override;
 
+  static void parse(const std::string & data_file_name,
+                    std::vector<Real> & x,
+                    std::vector<Real> & y,
+                    ColumnMajorMatrix & z);
```

Comment:
you can add a std::string parameter object_name which will just be populated by a call to name() on most objects to keep the informative error messages